src/python/pants/backend/python/util_rules/interpreter_constraints_test.py

```diff
@@ -293,105 +292,14 @@ def test_group_field_sets_by_constraints_with_unsorted_inputs() -> None:
 @pytest.mark.parametrize(
     "constraints,expected",
     (
-        # If it's a single constraint already, simply preserve it.
-        (["==2.7.3"], "==2.7.3"),
-        ([">=3.6,<4"], ">=3.6,<4"),
-        ([">=2.7.3,<2.7.6,!=2.7.4"], ">=2.7.3,<2.7.6,!=2.7.4"),
-        # Python 2.7 only.
-        (["==2.7.13", "==2.7.14"], ">=2.7.13,<=2.7.14"),
-        (["==2.7.13", "==2.7.14", "==2.7.15"], ">=2.7.13,<=2.7.15"),
-        (["==2.7.13", ">=2.7.15,<3"], ">=2.7.13,<2.8,!=2.7.14"),
-        (["==2.7.*", ">=2.7.15,<2.8"], ">=2.7,<2.8"),
-        (["==2.7.*,!=2.7.1", ">=2.7.15,<2.8"], ">=2.7,<2.8,!=2.7.1"),
-        # Python 3, single minor version only.
-        (["==3.6.8", "==3.6.9"], ">=3.6.8,<=3.6.9"),
-        (["==3.6.13", ">=3.6.15,<3.7"], ">=3.6.13,<3.7,!=3.6.14"),
-        (["==3.6.*", ">=3.6.15,<3.7"], ">=3.6,<3.7"),
-        (["==3.6.*,!=3.6.1", ">=3.6.15,<3.7"], ">=3.6,<3.7,!=3.6.1"),
-        # Python 3, multiple versions.
-        (["==3.6.*", "==3.7.*"], ">=3.6,<3.8"),
-        ([">=3.6.2,<3.7", "==3.7.*"], ">=3.6.2,<3.8"),
-        (["==3.6.*,!=3.6.5", "==3.7.*"], ">=3.6,<3.8,!=3.6.5"),
-        (["==3.6.*", "==3.8.*"], ">=3.6,<3.9,!=3.7.*"),
-        # Python 2.7 mixed w/ Python 3.
-        (["==2.7.*", ">=3.5,<4"], f">=2.7,<3.10,{_SKIPPED_PY3}"),
-        (["==2.7.*,!=2.7.18", ">=3.5,<4"], f">=2.7,<3.10,!=2.7.18,{_SKIPPED_PY3}"),
-        (["==2.7.*", ">=3.5,<3.6"], f">=2.7,<3.6,{_SKIPPED_PY3}"),
-        (["==2.7.*", ">=3.5,<3.5.4"], f">=2.7,<=3.5.3,{_SKIPPED_PY3}"),
-        (["==2.7.*", ">=3.5.2,<3.6"], f">=2.7,<3.6,{_SKIPPED_PY3},!=3.5.0,!=3.5.1"),
-        (["==2.7.*", ">=3.5.2,<3.5.4"], f">=2.7,<=3.5.3,{_SKIPPED_PY3},!=3.5.0,!=3.5.1"),
-        (
-            ["==2.7.13", "==3.5.3"],
-            (
-                ">=2.7.13,<=3.5.3,"
-                + ",".join(f"!=2.7.{p}" for p in range(14, _EXPECTED_LAST_PATCH_VERSION + 1))
-                + f",{_SKIPPED_PY3},!=3.5.0,!=3.5.1,!=3.5.2"
-            ),
-        ),
-        (["==2.7.*", ">=3.5,<3.6.4"], f">=2.7,<=3.6.3,{_SKIPPED_PY3}"),
-        (["==2.7.*", ">=3.5.2,<3.6.4"], f">=2.7,<=3.6.3,{_SKIPPED_PY3},!=3.5.0,!=3.5.1"),
-        (["==2.7.*", ">=3.5,<3.6.4,!=3.6.0"], f">=2.7,<=3.6.3,{_SKIPPED_PY3},!=3.6.0"),
-        (["==2.7.*", "==3.5.*", "==3.8.*"], f">=2.7,<3.9,{_SKIPPED_PY3},!=3.6.*,!=3.7.*"),
-        (
-            ["==2.7.*", "==3.5.*", ">=3.7.15,<3.8", "==3.8.*"],
-            (f">=2.7,<3.9,{_SKIPPED_PY3},!=3.6.*," + ",".join(f"!=3.7.{p}" for p in range(0, 15))),
-        ),
+        (["==2.7.*"], "==2.7.*"),
+        (["==2.7.*", ">=3.6,!=3.6.1"], "==2.7.* || !=3.6.1,>=3.6"),
+        ([], "*"),
+        (["CPython==2.7", "Cpython"], "*"),
```

Comment:
This test covers three different behaviours:
* How `CPython` is handled
* Case insensitivity (i.e. `CPython` == `Cpython`)
* Wildcards

I think the intention here would be better covered with three separate test cases

Reply:
Updated the test to better capture what I was after. The `CPython` piece was distracting - FYI `==2.7.*` is Pants shorthand for `CPython==2.7.*`, we automatically add that prefix.